front/lib/resources/content_fragment_resource.ts

```diff
@@ -365,13 +367,68 @@ async function renderFromFileId(
   } else {
     const content = await getProcessedFileContent(workspace, fileId);
 
+    const messageWithFullContent = {
+      role: "content_fragment" as const,
+      name: `inject_${contentType}`,
+      content: [
+        {
+          type: "text",
+          text: `<attachment type="${contentType}" title="${title}">\n${content}\n</attachment>`,
+        } as Content,
+      ],
+    };
+
+    if (contentType !== "text/csv") {
+      return new Ok(messageWithFullContent);
+    }
+
+    // For CSV files we have a custom rule:
+    // We render completely the content if the csv is small or a snippet if the csv is big.
+    // A big CSV is a CSV for which tokens counts for more than 10% of the context size of the model.
+    const tokenSize = await tokenCountForText(content, model);
+    const maxTokens = model.contextSize * 0.1;
+
+    if (tokenSize.isErr()) {
+      logger.error(
+        { error: tokenSize.error },
+        "Failed to count tokens for CSV content. Returning full content."
+      );
+      return new Ok(messageWithFullContent);
```

Comment:
Why do we not error here ?

Reply:
Just because we have the content already so I assumed better log an error and render the conversation if we can. If it fails on renderConversationForModel it will error though. 

This is not strongly held so I can def change to return an error.